src/lib/network_pool/with_nonce.ml

```diff
@@ -0,0 +1,8 @@
+open Core_kernel
+
+[%%versioned
+module Stable = struct
+  module V1 = struct
+    type 'a t = { message : 'a; nonce : int } [@@deriving compare, sexp, yojson]
```

Comment:
Could you add a comment here describing why this is needed?

Reply:
Added the following comment:

>   Appending a nonce to messages emitted by transaction and snark pools allows to circumvent
  libp2p's gossip deduplication logic and allow broadcasts to be broadcasted over and over again.